pkg/installer/generateconfig.go

```diff
@@ -43,6 +43,8 @@ func (m *manager) generateInstallConfig(ctx context.Context) (*installconfig.Ins
 		return nil, nil, errors.WithStack(err)
 	}
 
+	os.Setenv("OPENSHIFT_INSTALL_INVOKER", "ARO")
```

Comment:
`generateInstallConfig()` is not called in the unit test, so the unit test here fails when run against the upstream installer branch.

The two places that are calling `Resolve()` to generate assets are in `applyInstallConfigCustomisations()`, and in the 'create manifests` command just after `applyInstallConfigCustomisations()` is run. And only the first of these ends up calling any installer code, because all the assets are created by that before the second one runs. So I think it is safe to call this only from `applyInstallConfigCustomisations()`, which would ensure it is testable.

However, a more future-proof place to put it would be in an `init()` function somewhere in the `pkg/installer` package.